spot_rl_experiments/spot_rl/real_policy.py

```diff
@@ -24,76 +29,160 @@ def to_tensor(v):
-        config.RL.POLICY.OBS_TRANSFORMS.ENABLED_TRANSFORMS = []
-        config.freeze()
-        config.RL.POLICY["init"] = False
-
-        self.policy = policy_class.from_config(
-            config=config,
-            observation_space=observation_space,
-            action_space=action_space,
-        )
-        print("Actor-critic architecture:", self.policy)
-        # Move it to the device
-        self.policy.to(self.device)
-
-        # Load trained weights into the policy
-        self.policy.load_state_dict(
-            {k[len("actor_critic.") :]: v for k, v in checkpoint["state_dict"].items()}
-        )
+            if isinstance(checkpoint_path, str):
+                checkpoint = torch.load(checkpoint_path, map_location="cpu")
+            else:
+                checkpoint = checkpoint_path
+
+            # Load the config
+            config = checkpoint["config"]
+
+            """ Disable observation transforms for real world experiments """
+            config.defrost()
+            config.RL.POLICY.OBS_TRANSFORMS.ENABLED_TRANSFORMS = []
+            config.freeze()
+            config.RL.POLICY["init"] = False
+
+            # Load the policy using policy class
+            self.policy = policy_class.from_config(
+                config=config,
+                observation_space=observation_space,
+                action_space=action_space,
+            )
+            # Move it to the device
+            self.policy.to(self.device)
+
+            # Load trained weights into the policy
+            self.policy.load_state_dict(
+                {
+                    k[len("actor_critic.") :]: v
+                    for k, v in checkpoint["state_dict"].items()
+                }
+            )
 
+        print("Loaded Actor-critic architecture")
         self.prev_actions = None
         self.test_recurrent_hidden_states = None
         self.not_done_masks = None
         self.config = config
         self.num_actions = action_space.shape[0]
         self.reset_ran = False
+        self.is_hab3_policy = is_hab3_policy
         print("Policy loaded.")
 
     def reset(self):
         self.reset_ran = True
-        self.test_recurrent_hidden_states = torch.zeros(
-            1,  # The number of environments. Just one for real world.
-            self.policy.net.num_recurrent_layers,
-            self.config.RL.PPO.hidden_size,
-            device=self.device,
-        )
+        if not self.is_hab3_policy:
+            self.test_recurrent_hidden_states = torch.zeros(
+                1,  # The number of environments. Just one for real world.
+                self.policy.net.num_recurrent_layers,
+                self.config.RL.PPO.hidden_size,
+                device=self.device,
+            )
+        else:
+            self.test_recurrent_hidden_states = torch.zeros(
+                1,  # The number of environments. Just one for real world.
+                self.config.get("MOBILE_GAZE_NUM_RECURRENT_LAYERS", 4),
```

Comment:
Can't store attributes in torchscipt model as we are using torch.jit.trace method instead of torch.jit. script method for converting pytorch model into torchscript. Currently torch.jit.script method is throwing error & thus without breaking anything, the simplest solution I came up with is to store these attributes in standard deviation's pth file.
I created a dictionary with std tensor & other required attributes before saving to .pth file (standard deviation tensor is loaded separately during inference).    @akshararai